### ISSUE TITLE
0.Machine Learning

### ISSUE THREAD

**Issue by pillarliang** [pillarliang/blog#3]
## What:

**Artificial intelligence** is an area of computer science, where the goal is to enable computers and machines to perform human-like tasks and simulate human behavior

**Machine learning** is turning things (data) into numbers and **finding patterns** in those numbers.

> **Note**: “patterns” is an arbitrary term, you’ll often hear “embedding”, “weights”, “feature representation”, “feature vectors” all referring to similar things.

<img width="610" alt="Untitled" src="https://github.com/pillarliang/blog/assets/28833267/904fa0cd-4122-4592-a1dc-5e3c02e3364f">

### **Types of Machine Learning**
<img width="500" alt="Untitled 1" src="https://github.com/pillarliang/blog/assets/28833267/6ebab6a1-2dc0-458c-8de3-e29903fa01ff">

**1. Supervised learning**
uses labeled inputs to train models and learn outputs

[Ensemble learning | 集成学习](https://www.notion.so/Ensemble-learning-1-70ee5c8983484ca292e7233c04baf663?pvs=21)
- 集成学习是把多个模型组合在一起，以提高性能。

**2. Unsupervised learning**
uses unlabeled datato train models and learn outputs

**3. Reinforcement learning**
Agent learn in interactive environment based on rewards and penalties

## Why:
For a complex problem, we can’t think of all the rules.

## How: 7 steps for machine learning
Machine learning is divided into 7 steps in the actual operation level:

### 2. Data preprocessing
Data Preparation and Data Transformation are equally as important as the modelling algorithm (arguably more important).
<img width="490" alt="Untitled 1" src="https://github.com/pillarliang/blog/assets/28833267/6ebab6a1-2dc0-458c-8de3-e29903fa01ff">

- **Dataset**
[Dataset](https://www.notion.so/Dataset-089e31ecf076484fbe4194ce0a06b1ae?pvs=21)

- **Data Cleaning**

- **Exploratory Data Analysis | EDA | 探索性数据分析**
[EDA | 探索性数据分析](https://www.notion.so/EDA-9916a1e8a222445aab6650c5675c313f?pvs=21)

- **Feature Engineering | 特征工程**
[Feature Engineering | 特征工程](https://www.notion.so/Feature-Engineering-2c1675aec139441e9a13466f481c88ea?pvs=21)

### 5. Evaluation
- [Loss and Error surface](https://www.notion.so/Loss-and-Error-surface-aa9bb98c6bc2475f900ae58d39366731?pvs=21)
- [Metrics of Performance](https://www.notion.so/Metrics-of-Performance-69371d7f6fce4745970aedd4a154c81c?pvs=21)

### 6. Parameter adjustment | Optimization
[Search & Optimization](https://www.notion.so/Search-Optimization-2d93d411def64cc3a89cf1e89ddf2ead?pvs=21)

- **超参数：**
[Grid Search](https://www.notion.so/Grid-Search-3691390c09964583b67e13e6ddb433e0?pvs=21)

- **参数：**
    - [**Normal Equation |** 正规方程](https://www.notion.so/Normal-Equation-c9991496ed014a60baeb9b0615b08e0e?pvs=21)
    - [Gradient Descent](https://www.notion.so/Gradient-Descent-33c4e07694dd49b3a08ab0e6b919496a?pvs=21)

### 7. Forecast (beginning)
- [prediction](https://www.notion.so/prediction-1b002d03b4d34594b4f40860d59e77f3?pvs=21)

## Lib
[Scikit Learn](https://www.notion.so/Scikit-Learn-349c963e198a438da65c23b44488b42f?pvs=21)
